app/services/users.js

```diff
@@ -2,20 +2,35 @@ const { User } = require('../models');
 const errors = require('../errors');
 const logger = require('../logger');
 const utils = require('../utils');
+const { passwordEncryption } = require('../utils');
+const { ADMIN_ROLE, REGULAR_ROLE } = require('../constants');
 
 exports.userRegister = user =>
-  User.create(user).catch(() => {
-    logger.info(`Error trying to create the user ${user.firstName} ${user.lastName}`);
-    throw errors.databaseError('Error processing request in database.');
-  });
+  exports
+    .findUserBy({ email: user.email })
+    .then(foundUser => {
+      if (foundUser) {
+        throw errors.badRequest('The email is already registered.');
+      }
+      return passwordEncryption(user);
+    })
+    .then(response =>
+      User.create(response).catch(() => {
```

Comment:
this not sounds good. a response seems a http response and you dont give a http response to the database to create an user. put the names clearer please